### PR TITLE
feat: add request header

### PR DESCRIPTION
**Pull request by spacewander**

### CODE REVIEW

File: src/http/ngx_http_wasm_api.c [hunk 1]
```diff
@@ -936,13 +936,20 @@ proxy_add_header_map_value(int32_t type, int32_t key_data, int32_t key_size,
...
-        if (rc != NGX_OK) {
-            return PROXY_RESULT_BAD_ARGUMENT;
-        }
+        break;
+
+    default:
+        return PROXY_RESULT_BAD_ARGUMENT;
     }
 
+
```
Comment (membphis) [inline src/http/ngx_http_wasm_api.c:952]: style: why we need two blank lines here?

**Review by membphis — APPROVED** [review]

File: t/http_req_header.t [hunk 2]
```diff
@@ -160,3 +160,41 @@ location /t {
...
+
+
+=== TEST 9: add header
+--- config
+location /t {
+    access_by_lua_block {
+        local wasm = require("resty.proxy-wasm")
+        local plugin = assert(wasm.load("plugin", "t/testdata/http_header/main.go.wasm"))
+        local ctx = assert(wasm.on_configure(plugin, 'req_hdr_add'))
+        assert(wasm.on_http_request_headers(ctx))
```
Comment (membphis) [inline t/http_req_header.t:173]: Q: why we need to call `on_http_request_headers` function?

Comment (membphis) [reply t/http_req_header.t:173]: I know the reason after read the doc: https://github.com/api7/wasm-nginx-module#on_http_request_headers

File: src/http/ngx_http_wasm_api.c (hunk 1 above)
Comment (spacewander) [reply src/http/ngx_http_wasm_api.c:952]: This is a common style of Nginx C code.

**Pull request merged**